app/liquid_tags/user_subscription_tag.rb

```diff
@@ -13,73 +13,70 @@ class UserSubscriptionTag < LiquidTagBase
 
     // Hiding/showing elements
     // ***************************************
-
-    const subscriptionSignedIn = document.getElementById('subscription-signed-in');
-    const subscriptionSignedOut = document.getElementById('subscription-signed-out');
-    const responseMessage = document.getElementById('response-message');
-    const subscriberAppleAuth = document.getElementById('subscriber-apple-auth');
-    const confirmationModal = document.getElementById('user-subscription-confirmation-modal');
-    const profileImageContainer = document.getElementById('profile-images');
-    const subscriberImageContainer = document.querySelector('.ltag__user-subscription-tag__subscriber-profile-image');
-
     function clearSubscriptionArea() {
-      subscriptionSignedIn.classList.add("hidden");
-      subscriptionSignedOut.classList.add("hidden");
-      responseMessage.classList.add("hidden");
-      subscriberAppleAuth.classList.add("hidden");
+      document.getElementById('subscription-signed-in')?.classList.add("hidden");
+      document.getElementById('subscription-signed-out')?.classList.add("hidden");
+      document.getElementById('response-message')?.classList.add("hidden");
+      document.getElementById('subscriber-apple-auth')?.classList.add("hidden");
```

Comment:
out of curiosity, how does this work if we do `style.display = 'none';` here instead of adding a class? i noticed the slight flicker from the `Subscribe` button to the `You are already subscribed.` text, and I wonder if adding a class is contributing to that 🤔

Reply:
It still flickers 😞  - it was originally that way with `display = 'none'`. The flicker is definitely something we should explore and improve. This is the first liquid tag with all these different "states" using async JS 🙈 .

Reply:
Hm, weird. Maybe I'll investigate it if I'm feeling adventurous during cool down! It's not major though, and if it was further down the page with actual _text_ in the article (as opposed to my 3 word article 😂), then it wouldn't even be noticeable!